tests/containers/validate_btrfs.pm

```diff
@@ -57,26 +58,30 @@ sub _test_btrfs_thin_partitioning {
     validate_script_output "btrfs fi df $dev_path", sub { m/^Data.+total=[1-9].*[KMG]iB, used=\d+.+[KMG]iB/ };
 }
 
+# Fill up the btrfs subvolume, check if it is full and then increase the available size by adding another disk
 sub _test_btrfs_device_mgmt {
     my ($rt, $dev_path) = @_;
     my $container  = 'registry.opensuse.org/cloud/platform/stack/rootfs/images/sle15';
     my $btrfs_head = '/tmp/subvolumes_saved';
     record_info "test btrfs";
     script_run("df -h");
     # Determine the remaining size of /var
-    my $var_free = script_output('df 2>/dev/null | grep /var | awk \'{print $4;}\'');
-    # /var is using its own partition. Create file in the container enough to fill up the partition up to 99%
-    my $fill = int($var_free * 0.99 * 1024);    # df returns the size in KiB
+    my $var_free   = script_output('df 2>/dev/null | grep /var | awk \'{print $4;}\'');
+    my $var_blocks = script_output('df 2>/dev/null | grep /var | awk \'{print $2;}\'');
+    # Create file in the container enough to fill the "/var" partition (where the container is located)
+    my $fill = int($var_free * 1024 * 0.99);    # df returns the size in KiB
     $rt->up('huge_image', keep_container => 1, cmd => "fallocate -l $fill bigfile.txt");
     validate_script_output "df -h --sync|grep var", sub { m/\/dev\/vda.+\s+(9[7-9]|100)%/ };
-    # partition should be full
+    # check if the partition is full
     my ($total, $used) = _btrfs_fi("/var");
-    die "partition should be full" unless (int($total) == int($used));
-    # Due to disk space this should fail
-    die("pull still works") if ($rt->pull("$container") == 0);
+    die "partition should be full" unless (int($used) >= int($total * 0.99));
+    die("pull should fail on full partition") if ($rt->pull("$container") == 0);
+    # Increase the amount of available storage by adding the second HDD ('/dev/vdb') to the pool
     assert_script_run "btrfs device add /dev/vdb $dev_path";
     assert_script_run "btrfs fi show $dev_path/btrfs";
-    validate_script_output "lsblk | grep vdb", sub { m/vdb.+20G/ };
+    validate_script_output "lsblk | grep vdb", sub { m/vdb.+[2-9][0-9]G/ };
+    my $var_blocks_after = script_output('df 2>/dev/null | grep /var | awk \'{print $2;}\'');
+    die "available number of block didn't increase" if ($var_blocks >= $var_blocks_after);
```

Comment:
I would print both `$var_blocks` and `$var_blocks_after` out in `record_info` for debugging.

Reply:
done